### ISSUE TITLE
TextBox should set IsPassword accessible property when it contains password

### ISSUE THREAD

**Issue by dmitrii-drobotov** [dotnet/winforms#7816]
### .NET version

8.0.100-alpha.1.22423.9

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Worked in .NET 6 but not in .NET 7, didn't check earlier versions

### Issue description

From [Edit docs](https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-supporteditcontroltype#relevant-properties) on [UIA_IsPasswordPropertyId](https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-automation-element-propids) property:

> Must be set to TRUE on edit controls that contain passwords. If an edit control does contain Password contents then this property can be used by a screen reader to determine whether keystrokes should be read out as the user types them.

There are `UseSystemPasswordChar` and `PasswordChar` properties on `TextBox` that allow formatting content as password.

### Current behavior:

When setting either `UseSystemPasswordChar` or `PasswordChar` properties on a `TextBox`, the `IsPassword` property stays `False`. 

![A screenshot of a form with 2 text boxes formatted as passwords. Accessibility Insights app shows that IsPassword property is false](https://user-images.githubusercontent.com/102954094/191551252-40912925-fcf4-45a7-b9d2-64699ad0134e.png)

Narrator announces all symbols entered into such text boxes.

### Expected behavior:

When setting either `UseSystemPasswordChar` or `PasswordChar` properties on a `TextBox`, the `IsPassword` property becomes `True`, and screen readers don't announce characters entered into it. 

### Steps to reproduce

Sample application:
[TextBoxPasswordSample.zip](https://github.com/dotnet/winforms/files/9627198/TextBoxPasswordSample.zip)

**Comment by Nora-Zhou01** [comment]
Verified this issue in the latest .Net 8.0 SDK build:8.0.100-alpha.1.22478.15,it was fixed.please check below screenshot:
![屏幕截图 2022-09-28 192225](https://user-images.githubusercontent.com/104609169/192925559-0151683c-9c10-4240-b276-d89b070cebe9.png)
